src/NewButton/styles.ts

```diff
@@ -227,6 +228,9 @@ export const getButtonStyles = (theme?: Theme) => {
     },
     '[data-component="trailingIcon"]': {
       gridArea: 'trailingIcon'
+    },
+    '[data-component="leadingIcon"] + [data-component="text"]': {
+      textAlign: 'left'
```

Comment:
Not sure if this leftAlign is required tbh.

Reply:
fair, how do we make it easier for the user to override this with left align?

this would be the override, not sure if requires knowing the internal details?

```css
[data-component="text"] {
  text-align: left
}
```

Reply:
True :( This is where composite components would be handy. Maybe such an override usecase means we want to open up the API?

Reply:
If we go with the center-aligned approach I recommended, we can get rid of this style block.